DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/Utility.cs

```diff
@@ -0,0 +1,172 @@
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Dicom;
+using EnsureThat;
+using Microsoft.Health.Dicom.DeID.SharedLib.Model;
+
+namespace Microsoft.Health.Dicom.Anonymizer.Core
+{
+    public class Utility
+    {
+        public static DateTimeOffset[] ParseDicomDate(DicomDate item)
+        {
+            EnsureArg.IsNotNull(item, nameof(item));
+
+            return item.Get<string[]>().Select(ParseDicomDate).ToArray();
+        }
+
+        public static DateTimeOffset ParseDicomDate(string date)
+        {
+            EnsureArg.IsNotNull(date, nameof(date));
+
+            return DateTimeOffset.ParseExact(date, "yyyyMMdd", CultureInfo.InvariantCulture);
+        }
+
+        public static DateTimeObject[] ParseDicomDateTime(DicomDateTime item)
+        {
+            EnsureArg.IsNotNull(item, nameof(item));
+
+            return item.Get<string[]>().Select(ParseDicomDateTime).ToArray();
+        }
+
+        public static DateTimeObject ParseDicomDateTime(string dateTime)
+        {
+            EnsureArg.IsNotNull(dateTime, nameof(dateTime));
+
+            Regex dateTimeRegex = new Regex(@"^((?<year>\d{4})(?<month>\d{2})(?<day>\d{2})(?<hour>\d{2})(?<minute>\d{2})(?<second>\d{2})(\.(?<millisecond>\d{1,6}))?(?<timeZone>(?<sign>-|\+)(?<timeZoneHour>\d{2})(?<timeZoneMinute>\d{2}))?)(\s*)");
+            var matches = dateTimeRegex.Matches(dateTime);
+            if (matches.Count != 1)
+            {
+                throw new Exception();
+            }
+
+            var groups = matches[0].Groups;
+
+            int year = groups["year"].Success ? int.Parse(groups["year"].Value) : 0;
+            int month = groups["month"].Success ? int.Parse(groups["month"].Value) : 0;
+            int day = groups["day"].Success ? int.Parse(groups["day"].Value) : 0;
+            int hour = groups["hour"].Success ? int.Parse(groups["hour"].Value) : 0;
+            int minute = groups["minute"].Success ? int.Parse(groups["minute"].Value) : 0;
+            int second = groups["second"].Success ? int.Parse(groups["second"].Value) : 0;
+            int millisecond = groups["millisecond"].Success ? int.Parse(groups["millisecond"].Value) : 0;
+
+            if (groups["timeZone"].Success)
+            {
+                int timeZoneHour = int.Parse(groups["timeZoneHour"].Value) * int.Parse(groups["sign"].Value + "1");
+                int timeZoneMinute = int.Parse(groups["timeZoneMinute"].Value);
+                return new DateTimeObject()
+                {
+                    DateValue = new DateTimeOffset(year, month, day, hour, minute, second, millisecond > 999 ? int.Parse(millisecond.ToString().Substring(0, 3)) : millisecond, new TimeSpan(timeZoneHour, timeZoneMinute, 0)),
```

Comment:
>millisecond > 999 ? int.Parse(millisecond.ToString().Substring(0, 3)) [](http://example.com/codeflow?start=91&length=69)

Is there real case that millisecond has an invalid value a(>999)?
If so, then how about other fields?
the regular expression can't ensure all fields values are valid.